homeassistant/components/fan/xiaomi_miio.py

```diff
@@ -216,28 +329,152 @@ def _try_command(self, mask_error, func, *args, **kwargs):
+            self._device.set_buzzer, True)
+
+    @asyncio.coroutine
+    def async_set_buzzer_off(self):
+        """Turn the buzzer off."""
+        if self.supported_features & SUPPORT_SET_BUZZER == 0:
+            return
+
+        yield from self._try_command(
+            "Turning the buzzer of the miio device off failed.",
+            self._device.set_buzzer, False)
+
+    @asyncio.coroutine
+    def async_set_child_lock_on(self):
+        """Turn the child lock on."""
+        if self.supported_features & SUPPORT_SET_CHILD_LOCK == 0:
+            return
+
+        yield from self._try_command(
+            "Turning the child lock of the miio device on failed.",
+            self._device.set_child_lock, True)
+
+    @asyncio.coroutine
+    def async_set_child_lock_off(self):
+        """Turn the child lock off."""
+        if self.supported_features & SUPPORT_SET_CHILD_LOCK == 0:
+            return
+
+        yield from self._try_command(
+            "Turning the child lock of the miio device off failed.",
+            self._device.set_child_lock, False)
+
+    # pylint: disable=no-self-use
+    @asyncio.coroutine
+    def async_set_led_on(self):
+        """Turn the led on."""
+        return
+
+    # pylint: disable=no-self-use
+    @asyncio.coroutine
+    def async_set_led_off(self):
+        """Turn the led off."""
+        return
+
+    # pylint: disable=no-self-use
+    @asyncio.coroutine
+    def async_set_favorite_level(self, level: int):
+        """Set the favorite level."""
+        return
+
+    # pylint: disable=no-self-use
+    @asyncio.coroutine
+    def async_set_led_brightness(self, brightness: int):
+        """Set the led brightness."""
+        return
+
+    # pylint: disable=no-self-use
+    @asyncio.coroutine
+    def async_set_target_humidity(self, humidity: int):
+        """Set the target humidity."""
+        return
+
+
+class XiaomiAirPurifier(XiaomiGenericDevice, FanEntity):
+    """Representation of a Xiaomi Air Purifier."""
+
+    def __init__(self, name, device, model):
+        """Initialize the plug switch."""
+        XiaomiGenericDevice.__init__(self, name, device, model)
+
+        self._state_attrs.update({
+            ATTR_MODEL: self._model,
+            ATTR_AIR_QUALITY_INDEX: None,
+            ATTR_TEMPERATURE: None,
+            ATTR_HUMIDITY: None,
+            ATTR_MODE: None,
+            ATTR_FILTER_HOURS_USED: None,
+            ATTR_FILTER_LIFE: None,
+            ATTR_FAVORITE_LEVEL: None,
+            ATTR_CHILD_LOCK: None,
+            ATTR_LED: None,
+            ATTR_MOTOR_SPEED: None,
+            ATTR_AVERAGE_AIR_QUALITY_INDEX: None,
+            ATTR_PURIFY_VOLUME: None,
+            ATTR_LEARN_MODE: None,
+            ATTR_SLEEP_TIME: None,
+            ATTR_SLEEP_LEARN_COUNT: None,
+            ATTR_EXTRA_FEATURES: None,
+            ATTR_TURBO_MODE_SUPPORTED: None,
+            ATTR_SLEEP_MODE: None,
+            ATTR_AUTO_DETECT: None,
+        })
```

Comment:
Would it make sense to have a list of these attrs and simply do `_state_attrs = {attr: None for attr in available_attributes}` and set the model attribute afterwards?